### PR TITLE
fix(core): manage the gorgone configuration files

### PR DESCRIPTION
**Pull request by sc979**
## Description

Manage the creation of mandatory folders and structural architecture
Needed for the browser part

**Fixes** # (none)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 2.8.x
- [ ] 18.10.x
- [ ] 19.04.x
- [ ] 19.10.x
- [x] 20.04.x (master)

### CODE REVIEW

File: libinstall/functions [hunk 1]
```diff
@@ -1271,6 +1287,85 @@ copy_ssh_keys_to_gorgone() {
...
+    ## create the required gorgone configuration structure
+    if [ ! -d "$GORGONE_CONFIG/config.d" ] ; then
+        conf_folder="$GORGONE_CONFIG/config.d"
+        mkdir -p "$conf_folder"
+        if [ $? -ne 0 ] ; then
+            echo_failure "$(gettext "Cannot create the subfolder") $conf_folder" "$fail"
+            return 1
+        fi
+    fi
+    log "INFO" "GORGONE_CONFIG/config.d: $GORGONE_CONFIG/config.d"
```
Comment (sc979) [inline libinstall/functions]:
```suggestion
    log "INFO" "GORGONE_CONFIG/config.d: $conf_folder"
```

File: libinstall/functions [hunk 2]
```diff
@@ -1271,6 +1287,85 @@ copy_ssh_keys_to_gorgone() {
     return 0
 }
 
+#----
+## Create gorgone configuration folders
+#----
+create_gorgone_configuration_structure() {
+    ## create the required gorgone configuration structure
+    if [ ! -d "$GORGONE_CONFIG/config.d" ] ; then
+        conf_folder="$GORGONE_CONFIG/config.d"
```
Comment (sc979) [inline libinstall/functions]:
```suggestion
    conf_folder="$GORGONE_CONFIG/config.d"
    if [ ! -d "$conf_folder" ] ; then
```

File: libinstall/functions [hunk 3]
```diff
@@ -1271,6 +1287,85 @@ copy_ssh_keys_to_gorgone() {
...
+configuration: !include config.d/*.yaml
+__EOT__
+
+            echo_success "Create centreon config.yaml" "$ok"
+        fi
+    fi
+    echo_success "$(gettext "Gorgone configuration structure")" "$ok"
+    return 0
+}
+
```
Comment (sc979) [inline libinstall/functions]:
```suggestion
create_gorgone_configuration_structure() {
    ## create the required gorgone configuration structure
    conf_folder="$GORGONE_CONFIG/config.d"
    if [ ! -d "$conf_folder" ] ; then
        mkdir -p "$conf_folder"
        if [ $? -ne 0 ] ; then
            echo_failure "$(gettext "Cannot create the subfolder") $conf_folder" "$fail"
            return 1
        fi
    fi
    log "INFO" "GORGONE_CONFIG/config.d: $conf_folder"

    # gorgone's config.yaml
    destination="$GORGONE_CONFIG/config.yaml"
    if [ -e $destination ] ; then
        yes_no_default "$(gettext "The configuration is already created. Do you want to overwrite it ?")"
        if [ $? -ne 0 ] ; then
            echo_info "$(gettext "If needed, please update") $destination" "$passed"
        else
            ${CAT} <<- __EOT__ > $destination
name: config.yaml
description: Configuration brought by Centreon Gorgone
configuration: !include $conf_folder/*.yaml
__EOT__

            echo_success "$(gettext "Create gorgone config.yaml")" "$ok"
        fi
    fi

    # gorgone's config.d/30-centreon.yaml
    destination="$conf_folder/30-centreon.yaml"
    if [ -e $destination ] ; then
        yes_no_default "$(gettext "The configuration is already created. Do you want to overwrite it ?")"
        if [ $? -ne 0 ] ; then
            echo_info "$(gettext "If needed, please update") $destination" "$passed"
        else
            ${CAT} <<- __EOT__ > $destination
name: centreon.yaml
description: Configure Centreon Gorgone to work with Centreon Web.
centreon: !include $CENTREON_ETC/config.d/*.yaml
__EOT__

            echo_success "$(gettext "Create gorgone 30-centreon.yaml")" "$ok"
        fi
    fi

    ## create the required centreon configuration structure
    if [ ! -d "$CENTREON_ETC/config.d" ] ; then
        mkdir "$CENTREON_ETC/config.d"
        if [ $? -ne 0 ] ; then
            echo_failure "$(gettext "Cannot create the subfolder") $CENTREON_ETC/config.d" "$fail"
            return 1
        fi
    fi
    log "INFO" "CENTREON_ETC/config.d: $CENTREON_ETC/config.d"

    # centreon's config.yaml
    destination="$CENTREON_ETC/config.yaml"
    if [ -e $destination ] ; then
        yes_no_default "$(gettext "The configuration is already created. Do you want to overwrite it ?")"
        if [ $? -ne 0 ] ; then
            echo_info "$(gettext "If needed, please update") $destination" "$passed"
        else
            ${CAT} <<__EOT__ > $destination
name: config.yaml
description: Configuration for Central server
configuration: !include config.d/*.yaml
__EOT__

            echo_success "$(gettext "Create centreon config.yaml")" "$ok"
        fi
    fi
    echo_success "$(gettext "Gorgone configuration structure")" "$ok"
    return 0
}

```

File: libinstall/functions [hunk 4]
```diff
@@ -1249,19 +1265,19 @@ copy_ssh_keys_to_gorgone() {
...
 
-    ${CHOWN} -R $GORGONE_USER:$GORGONE_GROUP $GORGONE_VARLIB/.ssh
+    ${CHOWN} -R "$GORGONE_USER:$GORGONE_GROUP" "$GORGONE_VARLIB/.ssh"
     if [ $? -ne 0 ] ; then
         echo_failure "$(gettext "Cannot modify the owner of the files in $GORGONE_VARLIB/.ssh folder to $GORGONE_USER:$GORGONE_GROUP")" "$fail"
         return 1
     fi
 
-    ${CHMOD} 600 $GORGONE_VARLIB/.ssh/id_rsa
+    ${CHMOD} "600" "$GORGONE_VARLIB/.ssh/id_rsa"
```
Comment (kduret) [inline libinstall/functions]:
```suggestion
    ${CHMOD} 600 "$GORGONE_VARLIB/.ssh/id_rsa"
```

File: libinstall/functions [hunk 5]
```diff
@@ -1271,6 +1287,85 @@ copy_ssh_keys_to_gorgone() {
...
+        mkdir -p "$conf_folder"
+        if [ $? -ne 0 ] ; then
+            echo_failure "$(gettext "Cannot create the subfolder") $conf_folder" "$fail"
+            return 1
+        fi
+    fi
+    log "INFO" "GORGONE_CONFIG/config.d: $conf_folder"
+
+    # gorgone's config.yaml
+    destination="$GORGONE_CONFIG/config.yaml"
```
Comment (ganoze) [inline libinstall/functions]: Gorgone's config.yaml should already be installed by Gorgone installer. Why do you care about it here ? IMHO lines 1305 to 1320 should be removed.

Comment (sc979) [reply libinstall/functions]: You're right. I'll check again

**Pull request merged**